docs/generating-code-from-schema.md

```diff
@@ -230,30 +230,139 @@ Finally, `maxProjectionDepth` will instruct codegen to stop generating beyond 2
-| includeMutations | Generate Query API only for the given list of Mutation fields | All mutations defined in schema |
-| includeSubscriptions | Generate Query API only for the given list of Subscription fields | All subscriptions defined in schema |
-| skipEntityQueries | Disable generating Entity queries for federated types | false |
-| shortProjectionNames | Shorten class names of projection types. These types are not visible to the developer. | false |
-| maxProjectionDepth | Maximum projection depth to generate. Useful for (federated) schemas with very deep nesting | 10 |
+| generatedSourcesDir | Build directory for Gradle                                                                                                                                                                  | build |
+| outputDir | Sub directory of the `generatedSourcesDir` to generate into                                                                                                                                 | generated |
+| exampleOutputDir | Directory to generate datafetcher example code to                                                                                                                                           | generated-examples |
+| includeQueries | Generate Query API only for the given list of Query fields                                                                                                                                  | All queries defined in schema |
+| includeMutations | Generate Query API only for the given list of Mutation fields                                                                                                                               | All mutations defined in schema |
+| includeSubscriptions | Generate Query API only for the given list of Subscription fields                                                                                                                           | All subscriptions defined in schema |
+| skipEntityQueries | Disable generating Entity queries for federated types                                                                                                                                       | false |
+| shortProjectionNames | Shorten class names of projection types. These types are not visible to the developer.                                                                                                      | false |
+| maxProjectionDepth | Maximum projection depth to generate. Useful for (federated) schemas with very deep nesting                                                                                                 | 10 |
+| includeImports             | Maps the custom annotation type to the package, the annotations belong to                                                                                                                   |                                                       |
```

Comment:
Would be good to specify that both includeImports and includeEnumImports are applicable only when used along with generateCustomAnnotations.

That said, does it even make sense to not have those settings specified by the user when using generateCustomAnnotations?

Reply:
@srinivasankavitha Yes you are right. this is only needed when generateCustomAnnotations are enabled. If it is not present and generateCustomAnnotations are enabled, as per current implementation, it will use an empty string for the annotation packages.